File: majority_element.cpp
class Solution {
public:
    int majorityElement(std::vector<int>& nums) {
        int a = 0;
        std::unordered_map<int, int> hm;
        for (int i = 0; i < nums.size(); i++) {
            if (hm.find(nums[i]) != hm.end()) {
                hm[nums[i]]++;
            } else {
                hm[nums[i]] = 1;
            }
        }
        for (int i = 0; i < nums.size(); i++) {
            if (hm[nums[i]] > nums.size() / 2) {
                a = nums[i];
            }
        }
        return a;
    }
};

File: majority_element.js
class Solution {
	/** 
	 * @param {number[]} nums
	 * @return {number}
	 */
	majorityElement(nums) {
		// Your code goes here
        let countMap = new Map();
        nums.forEach((num)=>{
            countMap.set(num,countMap.get(num)?countMap.get(num)+1:1);
        })
        let target = nums.length/2;
       for(let [key,value] of countMap){
           if(value>target){
               return key;
           }
       }
       return -1;
	}
};
